primer/brand
# Updates to compiled stylesheet

## Summary

Addresses multiple issues in the compiled stylesheet, including:

- Removal of escaped characters in selectors where modern CSS features like `:has` are used
- Switched to use CSS Logical Properties in compiled output, which is both necessary to fix the above and is now worth doing as the feature is widely available in modern browsers.
 
**Compiled CSS output size is now 9% lower than before.**

Fixes https://github.com/primer/brand/issues/823

## List of notable changes:



- Updates relevant postcss dependency in order to prevent rewriting of `:has` selectors. Previous versions are deprecated, so an upgrade was needed.
- Adds rule to preserve source usage of `:has` and logical properties

## What should reviewers focus on?



- Diff the current production bundle with the canary equivalent

## Steps to test:



1. Copy [this](https://unpkg.com/@primer/react-brand@latest/lib/css/main.css) (current) and enter into left hand side of a diff checker
1.  Copy [this](https://unpkg.com/@primer/react-brand@0.0.0-20241119234629/lib/css/main.css) (canary) and enter into right hand side of a diff checker
1. Manually compare the output and verify that only changes are:
  - direct use of logical properties
  - escaped chars have been removed
  - some other legacy rules have been removed, but nothing that is relevant to modern browsers

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">


![Screenshot 2024-11-22 at 11 59 59](https://github.com/user-attachments/assets/43cbd7ed-f0b3-4e3c-b2f7-ab6c50469a96)

 </td>
<td valign="top">


![Screenshot 2024-11-22 at 11 59 06](https://github.com/user-attachments/assets/975c96c1-e102-4e0a-97a4-ab9c8c7a5e1a)

</td>
</tr>
</table>
